R/ggplotly.R

```diff
@@ -702,6 +702,7 @@ gg2list <- function(p, width = NULL, height = NULL, tooltip = "all",
   )
   
   # if theme(legend.position = "none") is used, don't show a legend _or_ guide
+  npscales$scales <- Filter(function(x) x$guide != "none", npscales$scales)
   if (npscales$n() == 0 || identical(theme$legend.position, "none")) {
```

Comment:
How about instead handling the discrete scale by updating [this line](https://github.com/ropensci/plotly/blob/master/R/layers2traces.R#L75) to something like
```
sc$is_discrete() && sc$guide != "none"
```